### PR TITLE
Issue 158 ann multilayer doc improvements

### PR DESCRIPTION
**Pull request by mrn-aglic**
Improved docs for ANN Multilayer network. Issue opened at: https://github.com/NetLogo/models/issues/158

### CODE REVIEW

**Comment by nicolaspayette** [comment]
Thanks a ton, @mrn-aglic! I will review this shortly!

File: Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo [hunk 1]
```diff
@@ -141,6 +141,9 @@ to back-propagate
   let answer target-answer
 
   ask output-node-1 [
+    ;; activation * (1 - activation) is used because it is the derivation
+    ;; of the sigmoid activation function. If we used a different activation function
+    ;; we would use its derivation
```
Comment (SethTisue) [inline Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo]: "derivative"?

File: Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo [hunk 2]
```diff
@@ -436,7 +439,23 @@ The activation values of the input nodes are the inputs to the network. The acti
...
+In case we wished to use a different activation function, we would use it's derivation in calculating the local gradient.
+
+For each hidden node, the local gradient is calculated as follows:
+1. For each output node connected to the hidden node, multiply its local gradient with the weight of the link connecting them
+2. Sum all of the results from 1
+3. Multiply the result from 2 with the result of passing the activation value of the hidden node to the derivation of the activation function.
+
+To update the weights of each of the links we first calculate the multiplication of the learning rate with the local gradient of end2 (this will be the output-node in case the link connects a hidden node with the output node) and the activation value of end1 (this will be the hidden node in case the link connects a hidden node with the output node). The result is then added to the old weight.
+
+The propagate and back-propage phases are repeated for each training data (or example) introduced to the network.
```
Comment (SethTisue) [inline Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo]: typo: back-propage

File: Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo (hunk 1 above)
Comment (mrn-aglic) [inline Sample Models/Computer Science/Artificial Neural Net - Multilayer.nlogo]: Fixed the mistakes I noticed

**Comment by nicolaspayette** [comment]
I will make minor tweaks, but this looks pretty good. Thanks again, @mrn-aglic: this is truly appreciated!

**Pull request merged**